public/app/features/dashboard-scene/scene/layout-tabs/TabItemRenderer.tsx

```diff
@@ -21,15 +22,30 @@ export function TabItemRenderer({ model }: SceneComponentProps<TabItem>) {
   return (
     <Tab
       className={cx(
+        styles.container,
         isSelected && 'dashboard-selected-element',
         isSelectable && !isSelected && 'dashboard-selectable-element'
       )}
       active={isActive}
       role="presentation"
+      title={titleInterpolated}
       href={href}
       aria-selected={isActive}
       onPointerDown={onSelect}
       label={titleInterpolated}
     />
   );
 }
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  container: css({
+    maxWidth: theme.spacing(40),
+
+    '& > :is(a,button)': css({
+      textOverflow: 'ellipsis',
+      whiteSpace: 'nowrap',
+      wordBreak: 'break-word',
+      overflow: 'hidden',
+    }),
+  }),
+});
```

Comment:
I think this should probably (in some form, or optional via prop) be part of the Tab component

Reply:
Added `truncate` prop to Tab ^^